drivers/dac/max22007/max22007.c

```diff
@@ -0,0 +1,598 @@
+
+	if (!dev || (ch > (MAX22007_NUM_CHANNELS - 1)))
+		return -EINVAL;
+
+	reg_val = no_os_field_prep(DAC_CH_DATA_MASK, data);
+
+	return max22007_reg_write_msk(dev, MAX22007_CHANNEL_DATA_REG(ch), reg_val,
+				      DAC_CH_DATA_MASK);
+}
+
+/**
+ * @brief Read data from a DAC channel data register
+ * @param dev - MAX22007 device descriptor.
+ * @param ch - Channel number (0-3).
+ * @param data - Pointer to store the 16-bit data read from the channel.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22007_read_channel_data(struct max22007_dev *dev, uint8_t ch,
+			       uint16_t *data)
+{
+	if (!dev || !data || (ch > (MAX22007_NUM_CHANNELS - 1)))
+		return -EINVAL;
+
+	return max22007_reg_read(dev, MAX22007_CHANNEL_DATA_REG(ch), data);
+}
+
+/**
+ * @brief Write to LDAC register to update DAC output
+ * @param dev - MAX22007 device descriptor.
+ * @param ch_mask - Channel mask.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22007_write_ldac(struct max22007_dev *dev, uint8_t ch_mask)
+{
+	uint16_t reg_val;
+
+	if (!dev || (ch_mask > 0xF))
+		return -EINVAL;
+
+	reg_val = no_os_field_prep(MAX22007_LD_CTRL_MASK, ch_mask);
+
+	return max22007_reg_write_msk(dev, MAX22007_CTRL_REG, reg_val,
+				      MAX22007_LD_CTRL_MASK);
+}
+
+/**
+ * @brief Write to Soft Reset Register
+ * @param dev - MAX22007 device descriptor.
+ * @param sw_clear - Data register reset control.
+ * @param sw_reset - Software reset control.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+static int max22007_soft_reset(struct max22007_dev *dev, bool sw_clear,
+			       bool sw_reset)
+{
+	uint16_t reg_val = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	reg_val = no_os_field_prep(MAX22007_SW_CLR_MASK, sw_clear) |
+		  no_os_field_prep(MAX22007_SW_RST_MASK, sw_reset);
+
+	return max22007_reg_write(dev, MAX22007_SOFT_RESET_REG, reg_val);
+}
+
+/**
+ * @brief MAX22007 descriptor initialization function.
+ * @param device - MAX22007 device descriptor.
+ * @param init_param - Initialization parameter containing data about the device
+ * 		       descriptor to be initialized.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22007_init(struct max22007_dev **device,
+		  struct max22007_init_param init_param)
+{
+	struct max22007_dev *dev;
+	uint16_t reg_val;
+	uint8_t ch;
+	int ret;
+
+	dev = (struct max22007_dev *)no_os_calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	no_os_crc8_populate_lsb(max22007_crc8_table, MAX22007_CRC8_POLYNOMIAL);
+
+	ret = no_os_spi_init(&dev->comm_desc, init_param.comm_param);
+	if (ret)
+		goto err_spi;
+
+	ret = max22007_soft_reset(dev, true, true);
+	if (ret)
+		goto err_spi;
+
+	ret = max22007_configure_crc(dev, init_param.crc_en);
+	if (ret)
+		goto err_spi;
+
+	ret = max22007_reg_read(dev, MAX22007_REV_ID_REG, &reg_val);
+	if (ret)
+		goto err_spi;
+
+	if ((reg_val != MAX22007_REV_ID_REV0) || (reg_val != MAX22007_REV_ID_REV1)) {
```

Comment:
This should always evaluate to true. In case you are looking for one of the 2 registers values, && would be the proper logical operation for the condition.